core/store.go

```diff
@@ -63,3 +69,23 @@ func Del(k string) bool {
 	}
 	return false
 }
+
+func FilterVals(pattern string) ([]filterValTuple, error) {
+	var filteredVals []filterValTuple
+	for key, value := range store {
+		val := value.Value.(string)
+		// look for a pattern match
+		// TODO: use regex or something light weight
+		ok, err := filepath.Match(pattern, val)
+		if err != nil {
+			return []filterValTuple{}, err
+		}
```

Comment:
we are only doing wildcard match. this is an expensive operation again. please write a simple wildcard matcher and then we will extend the solution if needed.